### ISSUE TITLE
0.89 fails with git error on CI

### ISSUE THREAD

**Issue by jeremy-ebler-vineti** [returntocorp/sgrep-action#569]
We have the Semgrep agent docker image running on our Jenkins CI, until recently it was working correctly.

Here's the backtrace for `v1` / `semgrep 0.89.0 on python 3.10.4`:

```
Digest: sha256:3c41ae90baca4dd423d22dd7b8170883a686ee05df4428754b5433c56c4e0c95
Status: Image is up to date for returntocorp/semgrep-agent:v1
=== Running: SEMGREP_TIMEOUT="1800" SEMGREP_BASELINE_COMMIT="origin/develop" semgrep ci
Scan environment:
  versions    - semgrep 0.89.0 on python 3.10.4
  environment - running in environment git, triggering event is pull_request
  semgrep.dev - authenticated as REDACTED

Fetching configuration from semgrep.dev
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/semgrep/commands/ci.py", line 297, in ci
    scan_handler.start_scan(metadata.to_dict())
  File "/usr/local/lib/python3.10/site-packages/semgrep/meta.py", line 128, in to_dict
    commit_title = subprocess.check_output(
  File "/usr/local/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'show', '-s', '--format=%B']' returned non-zero exit status 128.
Could not start scan Command '['git', 'show', '-s', '--format=%B']' returned non-zero exit status 128.
```

The last successful run, `semgrep 0.87.0 on python 3.10.4` [bence-0.87.0](https://hub.docker.com/layers/semgrep-agent/returntocorp/semgrep-agent/bence-0.87.0/images/sha256-5df13ff8677d279fb51e0d24ced149541fb8018b6d7aa522f96d802e9b9bb70a?context=explore):

```
=== Running: SEMGREP_TIMEOUT=1800 SEMGREP_BASELINE_COMMIT=origin/develop semgrep ci
Scan environment:
  versions    - semgrep 0.87.0 on python 3.10.4
  environment - running in environment git, triggering event is pull_request
  semgrep.dev - authenticated as REDACTED

Fetching configuration from semgrep.dev
Fetching rules from https://semgrep.dev/registry.

Scanning across multiple languages:
    <multilang> | 65 rules × 52 files
             go | 45 rules ×  8 files

  Current version has 7 findings.

Switching repository to baseline commit 'origin/develop'.
  Will report findings introduced by these commits:
    * f8d465bc feat: ginkgo2 upgrade and code coverage reports

Scanning 2 files with 7 <multilang> rules.

Some files were skipped.
  Scan was limited to files changed since baseline commit.
  Scan skipped: 2 files larger than 1.0 MB, 349 files matching .semgrepignore patterns
  For a full list of skipped files, run semgrep with the --verbose flag.

Ran 619 rules on 26 files: 0 findings.
Ran 42 blocking rules, 35 audit rules, and 542 internal rules used for rule recommendations.
Reporting findings to semgrep.dev ...
Success.
No findings so exiting with code 0
```

Here's a sample invocation:
`docker run --volume /var/lib/jenkins/workspace/REDACTED:/src --workdir /src --env SEMGREP_APP_TOKEN=**** --env SEMGREP_BASELINE_REF=origin/develop --env SEMGREP_BRANCH=REDACTED --env SEMGREP_COMMIT=REDACTED --env SEMGREP_JOB_URL=https://github.com/REDACTED --env SEMGREP_PR_ID=REDACTED --env 'SEMGREP_PR_TITLE=REDACTED' --env SEMGREP_REPO_NAME=REDACTED --env SEMGREP_REPO_URL=https://github.com/REDACTED --env SEMGREP_TIMEOUT=1800 --rm returntocorp/semgrep-agent:v1 semgrep-agent`

Is this a bug, or do we need to change our pipeline to upgrade?

**Comment by mogopz** [comment]
I'm seeing the same thing (running the latest agent in a container on Buildkite)

**Comment by jeremy-ebler-vineti** [comment]
I get the same error with `versions    - semgrep 0.90.0 on python 3.10.4`

**Comment by brendongo** [comment]
@jeremy-ebler-vineti @mogggggg not sure what is happening but I run ` docker run --volume ${pwd}:/src --workdir /src  --env SEMGREP_APP_TOKEN=REDACTED returntocorp/semgrep-agent:v1 ls -la /src` I find that the pwd was not copied over but when I type out PWD it works as expected. The error message you are seeing is because `git show` failed because pwd is not a valid git directory

Anyway: I would suggest changing the workflow to use the semgrep docker image. https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#buildkite

**Comment by jeremy-ebler-vineti** [comment]
@brendongo your command fails because the syntax is wrong. `--volume ${pwd}:/src` tries to expand the shell variable `pwd`, but shells don't typically set that. You can use the shell variable `PWD` with either `--volume ${PWD}:/src` or `--volume $PWD:/src`. Alternatively, you can use the shell command `pwd` by using `--volume $(pwd):/src` (note the parens).

---

I tested `returntocorp/semgrep-agent:v1` and `returntocorp/semgrep` with the command `git show -s --format=%B` and I get this error:
```
fatal: unsafe repository ('/src' is owned by someone else)
To add an exception for this directory, call:

	git config --global --add safe.directory /src
script returned exit code 128
```

Version 0.87 (`returntocorp/semgrep-agent:bence-0.87.0`) returns the git subject line as expected.

Maybe the version of git was bumped, and that's why we're seeing this new behavior. 

A `ls -la /src` shows that the files mounted into the container are owned by `1001:1001`
```
drwxrwxr-x   10 1001     1001          4096 May  6 20:01 .
drwxr-xr-x    1 root     root          4096 May  6 20:05 ..
-rw-rw-r--    1 1001     1001           157 May  6 20:01 .dockerignore
-rw-rw-r--    1 1001     1001          1242 May  6 20:01 .env.example
drwxrwxr-x    8 1001     1001          4096 May  6 20:01 .git
# ...
```

I'm guessing I need to get these files mounted as `root`, or build a custom image that includes their suggested command. It would be great if this was addressed upstream.

**Comment by mogopz** [comment]
Thanks for the replies, I've opened an issue in the main Semgrep repo ([link](https://github.com/returntocorp/semgrep/issues/5168)).

TL;DR you should be able to set the `GITHUB_WORKSPACE` env var with a path to your code as a workaround

**Comment by brendongo** [comment]
Ah yes that error looks familiar @jeremy-ebler-vineti (also thanks for the info on the pwd stuff) With https://github.blog/2022-04-12-git-security-vulnerability-announced/ git changed default behavior to fix a security issue and this new version is included in 0.89.0 onwards. We added logic to patch the behavior for GithubActions https://github.com/returntocorp/semgrep/pull/5044/files so you probably need to do similar before calling semgrep in buildkite (aka run `git config --global --add safe.directory DIR_NAME`

**Comment by brendongo** [comment]
Oh I see you figured exactly this out in the other repo so will close this and continue there.
